contrib/src/test/java/gov/nasa/jpl/aerie/contrib/models/SampledResourceTest.java

```diff
@@ -0,0 +1,31 @@
+package gov.nasa.jpl.aerie.contrib.models;
+
+import gov.nasa.jpl.aerie.merlin.framework.junit.MerlinExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import static gov.nasa.jpl.aerie.merlin.framework.ModelActions.delay;
+import static gov.nasa.jpl.aerie.merlin.protocol.types.Duration.SECOND;
+import static gov.nasa.jpl.aerie.merlin.protocol.types.Duration.SECONDS;
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(MerlinExtension.class)
```

Comment:
What does this `@ExtendWith` do?

Reply:
Here are the relevant docs:
> `@ExtendWith` [...] is used to register [extensions](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/extension/Extension.html) for the annotated test class

https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/extension/ExtendWith.html

`MerlinExtension` is a JUnit extension we provide that makes it possible to unit test mission models within a simulation environment. Every test method is wrapped in a call to `SimulationDriver.simulateTask` - which is how we're able to delay, emit events, and query cells in the test method below.

Reply:
> MerlinExtension is a JUnit extension we provide that makes it possible to unit test mission models within a simulation environment. Every test method is wrapped in a call to SimulationDriver.simulateTask - which is how we're able to delay, emit events, and query cells in the test method below.

Thanks! I'd found the JUnit docs you linked but that hadn't helped since they assume you're already familiar with JUnit extensions. This explains pretty much every question I had with the test 👍